### DISCUSSION TITLE
build error on local code ELIFECYCLE errno 1

### DISCUSSION THREAD

**Discussion by BernardA** [vercel/next.js#14965] [Help]
# Bug report
The app is running fine in development, but when attempting my first build it throws error below.
Except for indicating the route, the error message does not provide much of a clue on the cause of the error.

It seems to build most of the pages ( about 1200 at this stage ) and then throws the error. 

## Describe the bug

This is the error message in terminal:

       

> > Build error occurred
Error: Export encountered errors on following paths:
	/search/[city]/[rootCategory]/[parentCategory]/[category]
    at exportApp (/Users/quiamo/Sites/quiamo-next/node_modules/next/dist/export/index.js:22:1166)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async build (/Users/quiamo/Sites/quiamo-next/node_modules/next/dist/build/index.js:36:218)
Automatically optimizing pages .npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! next-app@0.1.0 build: `next build`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the next-app@0.1.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/quiamo/.npm/_logs/2020-07-08T08_09_18_275Z-debug.log

The contents of the log file mentioned above:
     
      

> 0 info it worked if it ends with ok
1 verbose cli [ '/usr/local/bin/node', '/usr/local/bin/npm', 'run', 'build' ]
2 info using npm@6.13.4
3 info using node@v12.16.0
4 verbose run-script [ 'prebuild', 'build', 'postbuild' ]
5 info lifecycle next-app@0.1.0~prebuild: next-app@0.1.0
6 info lifecycle next-app@0.1.0~build: next-app@0.1.0
7 verbose lifecycle next-app@0.1.0~build: unsafe-perm in lifecycle true
8 verbose lifecycle next-app@0.1.0~build: PATH: /usr/local/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/Users/quiamo/Sites/quiamo-next/node_modules/.bin:/Users/quiamo/Library/Python/3.7/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Apple/usr/bin:/Applications/MAMP/bin/php/php7.3.9/bin:/usr/local/sbin:$PATH:/usr/local/opt/mysql-client/bin:/usr/local/Cellar/node/11.7.0/bin:/usr/local/share/npm/bin:$PATH:$HOME/.symfony/bin:$PATH
9 verbose lifecycle next-app@0.1.0~build: CWD: /Users/quiamo/Sites/quiamo-next
10 silly lifecycle next-app@0.1.0~build: Args: [ '-c', 'next build' ]
11 silly lifecycle next-app@0.1.0~build: Returned: code: 1  signal: null
12 info lifecycle next-app@0.1.0~build: Failed to exec build script
13 verbose stack Error: next-app@0.1.0 build: `next build`
13 verbose stack Exit status 1
13 verbose stack     at EventEmitter.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:332:16)
13 verbose stack     at EventEmitter.emit (events.js:321:20)
13 verbose stack     at ChildProcess.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
13 verbose stack     at ChildProcess.emit (events.js:321:20)
13 verbose stack     at maybeClose (internal/child_process.js:1021:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:286:5)
14 verbose pkgid next-app@0.1.0
15 verbose cwd /Users/quiamo/Sites/quiamo-next
16 verbose Darwin 19.5.0
17 verbose argv "/usr/local/bin/node" "/usr/local/bin/npm" "run" "build"
18 verbose node v12.16.0
19 verbose npm  v6.13.4
20 error code ELIFECYCLE
21 error errno 1
22 error next-app@0.1.0 build: `next build`
22 error Exit status 1
23 error Failed at the next-app@0.1.0 build script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 1, true ]

## To Reproduce

See this repository: https://github.com/BernardA/quiamo-next
## Expected behavior

Build app. 

## System information

- OS: macOs 
- Browser (if applies) [e.g. chrome, safari]
- Version of Next.js: 9.4.4
- Version of Node.js: 12.16.0

**Comment by fineon** [comment]
Running into this error as well and tip to delete and reinstall node_modules doesn't work

**Reply by fineon** [reply]
Since this bug only appears when i run `next export` script. So I managed to fix this bug by removing it and specifying the next version in package.json (it was `next:"latest"` then i changed it to `next: "10.0.9"`). Also deleted the package-lock.json per online forum answers as well. Hope someone may find this useful.

**Reply by bayucandra** [reply]
Based on other answer, this one works for me:

- Delete `node_modules` directory
- Delete `package-lock.json`
- Do `npm install`
- Perhaps should convert `next:"latest"` to `next:"x.x.x"` necessary just like what @fineon said. Mine was not using `"latest"` as the version definition, hence no need to do that.
